File: ruby/word-count/word_count.rb
class Phrase
    attr_reader :word_count

    def initialize(input)
      @word_count = input.scan(/\b[\w']+\b/)
        .reduce(Hash.new(0)) do |word_count, word|
          word_count[word.downcase] += 1
          word_count
        end
    end
  end

File: elixir/word-count/lib/word_count.ex
defmodule WordCount do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t()) :: map
  def count(sentence) do
    sentence
    |> String.replace(~r/[^a-zA-Z0-9äöüÄÖÜß _-]/, "")
    |> String.replace("_", " ")
    |> String.downcase
    |> String.split
    |> Enum.reduce(%{}, fn x, acc -> Map.update(acc, x, 1, &(&1 + 1)) end)
  end
end
